src/validators/api/endpoints.py

```diff
@@ -0,0 +1,124 @@
+import asyncio
+import json
+from typing import Any
+
+from pydantic import TypeAdapter, ValidationError
+from starlette.requests import Request
+from starlette.responses import JSONResponse, Response
+from starlette.status import HTTP_400_BAD_REQUEST
+from sw_utils import is_valid_exit_signature
+from web3 import Web3
+
+from src.config.settings import settings
+from src.validators.api.schemas import ValidatorApproval
+from src.validators.database import NetworkValidatorCrud
+from src.validators.execution import get_latest_network_validator_public_keys
+from src.validators.tasks import (
+    get_available_validators_for_registration,
+    pending_validator_registrations,
+    register_and_remove_pending_validators,
+)
+from src.validators.typings import Validator, ValidatorsRegistrationMode
+
+
+async def get_validators(request: Request) -> Response:
+    deposit_data = request.app.state.deposit_data
+
+    validators = await get_available_validators_for_registration(
+        keystore=None, deposit_data=deposit_data, run_check_deposit_data_root=False
+    )
+    validators = [v for v in validators if v.public_key not in pending_validator_registrations]
+
+    # get next validator index for exit signature
+    latest_public_keys = await get_latest_network_validator_public_keys()
+    next_validator_index = NetworkValidatorCrud().get_next_validator_index(list(latest_public_keys))
+
+    return JSONResponse(
+        [
+            {'public_key': validator.public_key, 'index': index}
+            for index, validator in enumerate(validators, next_validator_index)
+        ]
+    )
+
+
+async def approve_validators(request: Request) -> Response:
```

Comment:
Maybe `submit_validators` or `register_validators`? The caller is not approving here. Lets rename also other `approvals` names in the code.